docs/DeveloperGuide.md

```diff
@@ -2,24 +2,138 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Add Function
 
+#### System Architecture of the Add Function<br/>
+
+![](team/Architect-digram.PNG)
+
+The Architecture Diagram given above explains the high-level design of the add command.<br/>
+
+Below we will discuss a quick overview of each component.
+
+* UI: The User Interface of the app<br/>
+* Add Command: The main logic command of the add function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+
+#### Design of the Add Function<br/>
+
+UML Class Diagram of the Add function:
+![](team/uml.PNG)<br/>
```

Comment:
The UML diagram looked a bit strange with the large blank spaces within the classes and the much smaller font used for the multiplicities. 

![image](https://user-images.githubusercontent.com/53657436/97386254-659da980-190e-11eb-8e84-b07ae86fb4d5.png)